modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java

```diff
@@ -209,11 +218,25 @@ private Map<String, CET> _getCETsMap(long companyId) {
 		return serviceRegistrationsMap;
 	}
 
-	private boolean _isInclude(CET cet, String keywords, String type) {
+	private boolean _isInclude(
+		CET cet, boolean excludeInstanceScopedCETs, String keywords,
+		String type) {
+
 		if (Validator.isNotNull(type) && !Objects.equals(type, cet.getType())) {
 			return false;
 		}
 
+		if (excludeInstanceScopedCETs &&
```

Comment:
:thinking: I don't think it makes sense to change the `CETManager` API by adding a new concept like scope if it is not going to be applied to all CX types.

Either we:
-  Promote the scope to be generalized to all CXs (and in that case it would have to be stored in the DB, not in the `typeSettings`) or
- We don't do anything to `CETManager` and implement the `excludeInstanceScopedCETs`  filtering at the higher level (that would be `CETItemSelectorViewDescriptor`).

Reply:
mmm I'm having a second thought, I believe you're right Iván. I misunderstood the role of this flag.

The `instance scope CX` is a concept that only applies to one type of CET. That's changes things, as it's not a general purpose flag. 

IINM, flag influences in 2 aspects:

1.  the way that GlobalJS entries are injected in the response, that is, rather than choosing the page where it'll be added, it's just added unconditionally. 
2.  filter a list of GlobalJS entries for user selection. Usually, filter by type is enough, but not in this case as they need a subset of the GlobalJS, those not marked as "instance scoped".

Essentially, this flag is no different from things we already have in other CX types. For example Editor Config Contributor CX type declares 3 properties that dictate when such entries are being executed to contribute configurations to a given editor instance. Difference is that we don't offer users a filtered list of Editor Config Contributor entries for selection.

I do agree this need does not have to be promoted into the framework itself: any additional filtering that depends on per-type specificities should not manifest in the main Manager API. 

I would defend this being added to the framework as long as one of the following conditions are met:

- The new characteristic is common to all CET types, with no exception.
- Framework provides an agnostic way of adding extra filtering to the CX, by accepting a consumer-provided predicate function that is applied to all items in a way that can't be done from outside. This way different consumers would implement specific filtering functions. 
 
Nevertheless, as this filtering is an "additional last step" after the list has been returned, I don't see any benefit of adding this to the framework either. A different thing would be if such predicate had to be applied in the middle of the query, but this does not seem the case: first filter by type, then exclude those with "instance scope".

To sum up, CETManager should not deal with this, as otherwise we would keep adding specific (per type) methods to a general-purpose interface.

Thiago, could you please try to filter the list of GlobalJS CX entries in the `CETItemSelectorViewDescriptor.getSearchContainer() `as Iván suggests? This looks a much better place for this, there's no need to change any API.

Thank you!

Reply:
BTW, you should also add a getter in the CX type to be able to do the filtering and so that everybody knows that field exits. That would be this interface -> https://github.com/liferay/liferay-portal/blob/90fdc04e1bac43119571af253a2b09b685a33234/modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/GlobalJSCET.java#L18, its implementation, etc.

Reply:
Hey guys. I believe you are right. I did this because we are going to add this concept to other types of client extensions. But i don't think it will be all of them. I will update the PR with the suggested approach